stat/test/wilcoxon_hyp.go

```diff
@@ -0,0 +1,190 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package test
+
+import (
+	"math"
+	"sort"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+func ensureDataConformance(x []float64, y []float64) {
+	if len(x) == 0 {
+		panic("x array length is zero")
+	}
+	if len(y) == 0 {
+		panic("y array length is zero")
+	}
+	if len(x) != len(y) {
+		panic("dimension mismatch in ensure data conformance")
+	}
+}
+
+func calculateDifferences(dst []float64, x []float64, y []float64) {
+	ensureDataConformance(x, y)
+	ensureDataConformance(x, dst)
+	floats.SubTo(dst, x, y)
+}
+
+func calculateAbsDifference(dst []float64, x []float64, y []float64) {
+	ensureDataConformance(x, y)
+	ensureDataConformance(x, dst)
+	for i := 0; i < len(x); i++ {
+		dst[i] = math.Abs(y[i] - x[i])
+	}
+}
+
+func rank(in []float64, out []float64) ([]float64, int) {
```

Comment:
```suggestion
func rank(in, out []float64) ([]float64, int) {
```

also, we should probably consider moving `out` to the 1st arg (to harmonize with other signatures)